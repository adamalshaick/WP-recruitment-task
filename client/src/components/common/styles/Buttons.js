import styled from "styled-components";

export const ButtonsCard = styled.span`
  background-color: white;
  display: flex;
  width: 500px;
  height: 50px;
  box-shadow: 0px 0px 5px 0px rgba(0, 0, 0, 0.25);
  position: relative;

  @media (min-width: 1000px) {
    left: 50%;
    transform: translateX(-50%);
  }
`;

export const Button = styled.a`
  line-height: 50px;
  text-decoration: none;
  color: black;
  width: 100%;
  text-align: center;
  transition: all 0.25s;
  &:hover {
    background-color: whitesmoke;
    border-bottom: blue solid 1px;
    color: blue;
  }
`;
