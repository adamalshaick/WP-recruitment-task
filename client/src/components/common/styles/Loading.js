import styled, { keyframes } from "styled-components";

const load = keyframes`
0% {
    opacity: 0;
  }
  50% {
    opacity: 0.5;
  }
  100% {
    opacity: 0;
  }
`;

export const Dots = styled.div`
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translateX(-50%);
  font-size: 4rem;
  opacity: 0.5;
  animation: ${load} 1.5s infinite;
`;
