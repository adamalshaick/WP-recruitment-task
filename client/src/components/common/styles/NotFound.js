import styled from "styled-components";
import { Link } from "react-router-dom";

export const NotFoundWrapper = styled.section`
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
`;

export const Button = styled(Link)`
  color: black;
  text-decoration: none;
  background-color: whitesmoke;
  padding: 1rem;
  border-radius: 5px;
  box-shadow: 0px 0px 5px 0px rgba(0, 0, 0, 0.25);
  margin: 1rem;
`;
