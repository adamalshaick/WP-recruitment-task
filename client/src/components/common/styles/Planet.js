import styled from "styled-components";

export const PlanetCard = styled.section`
  position: absolute;
  left: 50%;
  top: 60%;
  transform: translate(-50%, -50%);
  background-color: white;
  opacity: ${props => (props.active ? "1" : "0")};
  visibility: ${props => (props.active ? "visible" : "hidden")};
  transition: all 0.3s;
  padding: 2rem;
  box-shadow: 0px 0px 5px 0px rgba(0, 0, 0, 0.35);
  border-radius: 5px;
  z-index: 2;
`;

export const PlanetWrapper = styled.div`
  width: 100%;
  height: 100%;
  background-color: white;
  opacity: ${props => (props.active ? "0.5" : "0")};
  visibility: ${props => (props.active ? "visible" : "hidden")};
  position: fixed;
  top: 0;
  left: 0;
  filter: brightness(90%);
  transition: all 0.3s;
`;

export const CardHeader = styled.header`
  margin: 1rem;
  font-size: 20px;
  font-weight: 500;
`;
