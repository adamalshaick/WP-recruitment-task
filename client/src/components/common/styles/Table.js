import styled, { keyframes } from "styled-components";

const entry = keyframes`
0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
`;

export const Table = styled.table`
  box-shadow: 0px 0px 5px 0px rgba(0, 0, 0, 0.25);
  border-radius: 5px;
  background-color: white;
  width: 70%;
  position: relative;
  border-spacing: 0;
  margin-top: 4rem;
  margin-bottom: 2rem;
  animation: ${entry} 0.5s;
  min-height: 500px;

  @media (min-width: 1000px) {
    left: 50%;
    transform: translateX(-50%);
  }
`;

export const TableRow = styled.tr`
  height: 50px;
  opacity: ${props => (props.active ? "0.2" : "1")};
  transition: all 0.3s;
  &:hover {
    background-color: ${props => (props.active ? "white" : "whitesmoke")};
    cursor: ${props => (props.active ? "auto" : "pointer")};
  }
`;

export const TableHeader = styled.tr`
  height: 50px;
  background-color: black;
  color: white;
  font-weight: 500;
  font-size: 14px;
`;

export const TableElement = styled.td`
  border-bottom: lightgrey solid 1px;
  text-align: right;
  padding: 0 3rem;
  width: 25%;
`;

export const TableName = styled.td`
  padding: 1rem;
  border-bottom: lightgrey solid 1px;
`;
