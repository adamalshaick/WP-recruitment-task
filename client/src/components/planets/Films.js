import React from "react";
import fetchDataDetails from "../hoc/fetchDataDetails";

export const Films = ({ data }) => {
  return <p>{data.title}</p>;
};

export default fetchDataDetails(Films);
