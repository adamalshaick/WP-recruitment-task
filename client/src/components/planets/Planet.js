import React, { Component } from "react";
import { TableRow, TableElement, TableName } from "../common/styles/Table";
import { PlanetWrapper } from "../common/styles/Planet";
import PlanetInfo from "./PlanetInfo";
import PropTypes from "prop-types";

export class Planet extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showPlanetInfo: false
    };
  }

  onClick = e => {
    e.preventDefault();
    this.setState(prevState => ({
      showPlanetInfo: !prevState.showPlanetInfo
    }));
  };

  render() {
    return (
      <>
        <TableRow onClick={this.onClick} active={this.state.showPlanetInfo}>
          <TableName>{this.props.planet.name}</TableName>
          <TableElement>{this.props.planet.diameter}</TableElement>
          <TableElement>{this.props.planet.climate}</TableElement>
          <TableElement>{this.props.planet.gravity}</TableElement>
          <TableElement>{this.props.planet.population}</TableElement>
        </TableRow>
        <PlanetInfo
          active={this.state.showPlanetInfo}
          planet={this.props.planet}
        />
        <PlanetWrapper
          active={this.state.showPlanetInfo}
          onClick={this.onClick}
        />
      </>
    );
  }
}

Planet.propTypes = {
  planet: PropTypes.object.isRequired
};

export default Planet;
