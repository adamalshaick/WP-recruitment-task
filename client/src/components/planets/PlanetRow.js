import React from "react";
import { PlanetCard, CardHeader } from "../common/styles/Planet";
import Films from "./Films";
import Residents from "./Residents";
import PropTypes from "prop-types";

export const PlanetRow = ({ active, planet }) => {
  return (
    <div>dd</div>
    // <PlanetCard active={active}>
    //   <CardHeader>
    //     {planet.name} planet details
    //     <hr />
    //   </CardHeader>
    //   <div style={{ display: "flex" }}>
    //     <div style={{ margin: "1rem" }}>
    //       <p>rotation period: {planet.rotation_period}</p>
    //       <p>orbital period: {planet.orbital_period}</p>
    //       <p>diameter: {planet.diameter}</p>
    //       <p>climate: {planet.climate}</p>

    //       <p>
    //         films:
    //         {active
    //           ? planet.films.map(film => {
    //               return <Films function={film} />;
    //             })
    //           : null}
    //       </p>
    //     </div>
    //     <div style={{ margin: "1rem" }}>
    //       <p>surface water: {planet.surface_water}</p>
    //       <p>population: {planet.population}</p>
    //       <p>terrain: {planet.terrain}</p>
    //       <p>gravity: {planet.gravity}</p>
    //       <p>
    //         residents:
    //         {active
    //           ? planet.residents.map(resident => {
    //               return <Residents function={resident} />;
    //             })
    //           : null}
    //       </p>
    //     </div>
    //   </div>
    // </PlanetCard>
  );
};

// PlanetRow.propTypes = {
//   planet: PropTypes.object.isRequired,
//   active: PropTypes.bool.isRequired
// };

export default PlanetRow;
