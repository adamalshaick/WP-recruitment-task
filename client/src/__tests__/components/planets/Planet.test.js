import React from "react";
import { Planet } from "../../../components/planets/Planet";
import {
  PlanetWrapper,
  CloseButton
} from "../../../components/common/styles/Planet";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { TableRow } from "../../../components/common/styles/Table";

const Props = {
  planet: {}
};

const wrapper = shallow(<Planet {...Props} />);

describe("render component", () => {
  it("renders component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe("show planet info functionality", () => {
  it("toggles planet info state", () => {
    expect(wrapper.state().showPlanetInfo).toEqual(false);
    wrapper.find(TableRow).simulate("click", { preventDefault() {} });
    expect(wrapper.state().showPlanetInfo).toEqual(true);
    wrapper.find(PlanetWrapper).simulate("click", { preventDefault() {} });
    expect(wrapper.state().showPlanetInfo).toEqual(false);
  });
});
