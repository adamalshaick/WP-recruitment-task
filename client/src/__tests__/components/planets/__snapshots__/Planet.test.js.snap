// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders component 1`] = `
<Fragment>
  <styled.tr
    active={false}
    onClick={[Function]}
  >
    <styled.td />
    <styled.td />
    <styled.td />
    <styled.td />
    <styled.td />
  </styled.tr>
  <PlanetInfo
    active={false}
    onToggle={[Function]}
    planet={Object {}}
  />
  <styled.tr
    active={false}
    onClick={[Function]}
  />
</Fragment>
`;
